src/app/mod.rs

```diff
@@ -238,13 +243,29 @@ impl App {
         &self,
         profile: Profile,
         wallet_name: String,
+        external_id: Option<String>,
+        metadata: Option<serde_json::Value>,
     ) -> Result<String, BriaError> {
         let wallet = self
             .wallets
             .find_by_name(profile.account_id, wallet_name)
             .await?;
         let keychain_wallet = wallet.current_keychain_wallet(&self.pool);
         let addr = keychain_wallet.new_external_address().await?;
+
+        let new_address = NewAddress::builder()
+            .address_string(addr.to_string())
+            .keychain_id(keychain_wallet.keychain_id)
+            .kind(bitcoin::pg::PgKeychainKind::External)
+            .address_idx(addr.index)
+            .external_id(external_id)
```

Comment:
I've thought about this and let's set external_id = internal_id.to_string() in the case when none is provided. We want to have an index for lookup on `external_id` and allowing `None` makes that more complicated.